docs/contributing/contributing-code.md

```diff
@@ -230,3 +230,42 @@ On M1 macs some `pre-commit` tests may fail.
 If those checks are running in docker containers (such as `hadolint`) and exit with code 137, this means they are running into a memory limit.
 This is because for running x86_64 images on an M1 mac, docker will utilize an emulation with qemu that can cause higher memory consumption.
 To fix this, change the memory limit in Docker Desktop in the Resources section of the Preferences, 8.00GB should be sufficient.
+
+### Working with git submodules
+
+A submodule is a reference to a specific commit in another git
+repository. This results in a subfolder containing a checkout of that
+repository at that particular commit. Submodules have a reputation of
+being tricky to use. To minimize problems, make sure to follow these
+guidelines:
+
+* When checking out a new branch or commit, update the submodules
+  using the command `git submodule update --init --recursive`.
+  Adding a shortcut to your shell can be useful. The following is a
+  Bash function that lets you call `gitup`. It goes into your `~/.bashrc`:
+
+```bash
+gitup() {
```

Comment:
Why not a Git alias? I think you could define `git up` to do that.

Reply:
Yes, you can have a git alias for this: https://github.com/nmote/dotfiles/blob/420a361/.gitconfig#L19

Reply:
Also, we have an internal Notion page for Git tips. Maybe we should move them all here? Notably, I think these config lines also help a lot with dealing with submodules:

https://github.com/nmote/dotfiles/blob/420a361/.gitconfig#L44-L46

Reply:
Let's not bikeshed here. If you want to change this section, please post an edit I can approve with one click.